File: code/shared/StorageHelper.java
package shared;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class StorageHelper {

    // Save content to file (creates directories if needed)
    public static void saveToFile(String content, String filePath) throws IOException {
        Path path = Paths.get(filePath);
        Files.createDirectories(path.getParent());  // create parent dirs if not exist
        Files.writeString(path, content);
        System.out.println("StorageHelper: Saved file to " + filePath);
    }

    // Read content from file
    public static String readFromFile(String filePath) throws IOException {
        Path path = Paths.get(filePath);
        String content = Files.readString(path);
        System.out.println("StorageHelper: Read file from " + filePath);
        return content;
    }
}

File: code/shared/storage_helper.py
import os

class StorageHelper:
    @staticmethod
    def save_to_file(content, file_path):
        try:
            os.makedirs(os.path.dirname(file_path), exist_ok=True)
            with open(file_path, 'w') as f:
                f.write(content)
            print(f"StorageHelper: Saved file to {file_path}")
        except Exception as e:
            print(f"StorageHelper: Failed to save file - {e}")

    @staticmethod
    def read_from_file(file_path):
        try:
            with open(file_path, 'r') as f:
                content = f.read()
            print(f"StorageHelper: Read file from {file_path}")
            return content
        except Exception as e:
            print(f"StorageHelper: Failed to read file - {e}")
            return None
